### DISCUSSION TITLE
Make a Fluid kernel for Jupyter?

### DISCUSSION THREAD

**Discussion by RaoOfPhysics** [fluid-org/fluid-examples#4] [Ideas]
Instead of writing a Markdown parser for Fluid, @rolyp and I discussed the possibility of making a Fluid kernel for Jupyter. There already [exist several for different languages](https://github.com/jupyter/jupyter/wiki/Jupyter-kernels) and Jupyter provides [some instructions for how to make one](https://jupyter-client.readthedocs.io/en/stable/kernels.html).

This will allow Fluid to function within a larger ecosystem that developers and some tech-minded journalists might be familiar with.

This also means that we can use something like Quarto (see #5) to handle the publishing (into HTML articles or HTML presentations).

**Comment by rolyp** [comment]
[Marimo](https://marimo.io/) is a very nice-looking Python notebook, might be interesting to look at how easy it would be to adapt it to host Fluid instead. (It doesn’t look like it has a plug-in/kernel architecture like Jupyter’s, though.)

**Reply by RaoOfPhysics** [reply]
Looks really nice! Seems limited to Python, though. Still, lots of ideas we could possibly steal for Fluid.
